src/game/engine/buildingtype.cpp

```diff
@@ -595,6 +596,14 @@ BuildingTypeClass &BuildingTypeClass::As_Reference(BuildingType type)
     return g_BuildingTypes[type];
 }
 
+BuildingTypeClass *BuildingTypeClass::As_Pointer(BuildingType type)
+{
+    DEBUG_ASSERT(type != BUILDING_NONE);
```

Comment:
Again, shouldn't assert on something that the function has handling for... if you have handling, you expect those values to happen at some point.